File: gb-angular-frontend/src/app/interfaces/product-package.ts
import {Package} from './package';
import {EndTime} from './end-time';
import {ReviewOverallScore} from './review-overall-score';

export interface ProductPackage {
  product_id: number;
  title: string;
  creation_date: string;
  sell_style: string;
  description: string;
  image_urls: string;
  brand_id: number;
  material: string;
  delivery_date: string;
  delivery_method: string;
  delivery_region: string;
  min_threshold: number;
  stock_quantity: number;
  stock_status: string;
  status: string;
  cost: number;
  regular_price: number;
  sale_price: number;
  category_id: number;
  seller_id: number;
  seller_username: string;
  likes: number;
  display_position: number;
  main_image: string;
  brand_name: string;
  category_name: string
  grpbuy_start: string;
  grpbuy_end: string;
  grpbuy_quantity: number;
  grpbuy_regular_price: number;
  grpbuy_sale_price: number;
  grpbuy_status: string;
  sell_type: string
  pkg_grp_id: number;
  pkg_grp_name: string;
  pkg_creation_date: string;
  pkg_quantity: number;
  pkg_product_id: number;
  pkg_product_title: string;
  pkg_status: string;
  pkg_grp_image: string;
  pkg_grpbuy_start: string;
  pkg_grpbuy_end: string;
  pkg_grpbuy_duration: number;
  pkg_grpbuy_duration_type: string;
  pkg_min_threshold: number;
  pkg_end_time_counter: EndTime;
  wished: string;
  package_count: number;
  review_overall_score?: ReviewOverallScore;
  timer:any;
}

File: gb-spring-backend/src/main/java/com/egroupbuy/model/ProductPackage.java
package com.egroupbuy.model;

import lombok.*;

import java.sql.Date;
import java.sql.Timestamp;


@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class ProductPackage {
    int product_id;
    String title;
    Timestamp creation_date;
    String sell_style;
    String description;
    String image_urls;
    int brand_id;
    String material;
    Timestamp delivery_date;
    String delivery_method;
    String delivery_region;
    int min_threshold;
    int stock_quantity;
    String stock_status;
    String status;
    double cost;
    int category_id;
    int seller_id;
    String seller_username;
    int likes;
    double regular_price;
    double sale_price;
    int display_position;
    String main_image;
    String brand_name;
    String category_name;
    Timestamp grpbuy_start;
    Timestamp grpbuy_end;
    int grpbuy_quantity;
    double grpbuy_regular_price;
    double grpbuy_sale_price;
    String grpbuy_status;
    String sell_type;
    int pkg_grp_id;
    String pkg_grp_name;
    Date pkg_creation_date;
    int pkg_quantity;
    int pkg_product_id;
    String pkg_product_title;
    String pkg_status;
    String pkg_grp_image;
    Timestamp pkg_grpbuy_start;
    Timestamp pkg_grpbuy_end;
    int pkg_grpbuy_duration;
    String pkg_grpbuy_duration_type;
    int pkg_min_threshold;
    String wished;
    int package_count;
    ReviewOverallScore review_overall_score;
}
